File: rdict-iphone/Classes/DictionaryEntry.m
//
//  DictionaryEntry.m
//  RDict
//
//  Created by Stephen Bodnar on 28/05/09.
//  Copyright 2009 __MyCompanyName__. All rights reserved.
//

#import "DictionaryEntry.h"


@implementation DictionaryEntry
@synthesize word;
@synthesize entry;

- (id) initWithWord: (NSString *) word_arg andEntry:(NSString *) entry_arg {
	
	self.word = [[NSString alloc] initWithString: word_arg];
	self.entry = [[NSMutableString alloc] initWithString: entry_arg];
	return self;
}

-(void) htmlifyEntry {
	NSData *htmlData = [[NSData alloc] initWithContentsOfURL: [NSURL fileURLWithPath:[[NSBundle mainBundle] pathForResource:@"dictionary_js" ofType:@"html"]]];
	NSString *htmlString = [[NSString alloc] initWithData: htmlData encoding: NSUTF8StringEncoding];
	[self.entry insertString:htmlString atIndex: 0];
}

- (void)dealloc {
	[self.word release];
	[self.entry release];
	[super dealloc];
}

@end

File: rdict-android/src/com/amplio/rdict/DictionaryEntry.java
package com.amplio.rdict;

public class DictionaryEntry {
	
	public long id = -1;
	public String headword = null;
	public String contents = null;
	
	public DictionaryEntry(String headword, String contents) {
		this.id = -1;
		this.headword = headword;
		this.contents = contents;
	}
	
	public DictionaryEntry(long id, String headword, String contents){
		this.id = id;
		this.headword = headword;
		this.contents = contents;
	}
}
